hathor/stratum/stratum.py

```diff
@@ -418,11 +418,17 @@ def handle_request(self, method: str, params: Optional[Union[list, dict]], msgid
             if not self.manager.can_start_mining():
                 return self.send_error(NODE_SYNCING, msgid)
 
+        if not isinstance(params, dict):
+            self.log.error(f'expected dict, received {params}. Trying to convert...')
+            try:
+                params = dict(params)  # type: ignore[arg-type]
```

Comment:
This is a behavior change, is this actually fixing anything if it succeeds? For example `dict([[1,2]])` will result in `{1: 2}', so sometimes `dict(a: list)` successfully returns a dict. Before this change we didn't have this behavior.

If we don't have this `try` block and instead only have the cast that is inside the `except` block, the behavior is exactly the same but we'd have better logging.

Reply:
Nice catch!

Reply:
Removed in https://github.com/HathorNetwork/hathor-core/pull/1049/commits/1f595d81b608578dae609479fe62b02ee77d1eab